### DISCUSSION TITLE
How to stream Text Completion?

### DISCUSSION THREAD

**Discussion by tunyawat** [srcnalt/OpenAI-Unity#85] [Q&A]
After following the tutorial, I was able to retrieve results for streamed ChatCompletion under the HandleResponse method using the code below:

```
    private void HandleResponse(List<CreateChatCompletionResponse> responses)
    {
        text.text = string.Join("", responses.Select(r => r.Choices[0].Delta.Content));
    }

```
However, when it comes to TextCompletion, the structure is different, and I'm not sure how to extract the result. I tried using the following line but it didn't produce any output.

```
    private void HandleResponse(List<CreateCompletionResponse> responses)
    {
        text.text = string.Join("", responses.Select(r => r.Choices[0].Text));
    }
```

For clarity, here is the Choice structure:

```
public struct Choice
{
    public string Text { readonly get; set; }
    public int? Index { readonly get; set; }
    public int? Logprobs { readonly get; set; }
    public string FinishReason { readonly get; set; }
}
```

**Comment by srcnalt** [comment]
Hi @tunyawat, only the Chat Completion endpoint had streaming support last time I worked on the project.
Also seems like OpenAI is deprecating text completion endpoint: https://platform.openai.com/docs/api-reference/completions

**Reply by tunyawat** [reply]
I'm curious as to why OpenAI just released the gptInstruct model and decided to deprecate the text completion endpoint.

I've utilized text completion because, as showed in one of your tutorial videos, interactions with NPCs become much more natural. Given that, I guess I'll work on refining the prompts to optimize results from chat completion then.
